docs/mkdocs/docs/api/macros/json_diagnostic_positions.md

```diff
@@ -6,9 +6,13 @@
 
 This macro enables position diagnostics for generated JSON objects.
 
-When enabled, two new properties: `start_pos()` and `end_pos()` are added to `nlohmann::json` objects and fields. `start_pos()` returns the start position of that JSON object/field in the original string the object was parsed from. Likewise, `end_pos()` returns the end position of that JSON object/field in the original string the object was parsed from.
+When enabled, two new properties: `start_pos()` and `end_pos()` are added to `nlohmann::json` objects and fields. `start_pos()` returns the start position of
+that JSON object/field in the original string the object was parsed from. Likewise, `end_pos()` returns the end position of that JSON object/field in the
+original string the object was parsed from.
 
-For objects and arrays, the start and end positions represent the positions of the opening and closing braces or brackets, respectively. For fields, the start and end positions represent either the opening and closing quotes for that field's value or the first and character after last in the field's numerical or predefined true/false/null values.
+For objects and arrays, the start and end positions represent the positions of the opening and closing braces or brackets, respectively. For fields, the start
+and end positions represent either the opening and closing quotes for that field's value or the first and character after last in the field's numerical or
```

Comment:
Now that I read this, this seems odd to me.  The end for array and object is the last character that represents the array or object, but for a value it is the character after the last?  Is that how it's implemented?  Is it intended to be implemented that way?

Reply:
Good point catching this. The documentation is slightly off - we always return one past the end of the substring. I will update the doc here and add a comment to the code as well.

Reply:
Updated